File: section-1/ds/strings/strings.js
let name = "Bro code";

// .charAt()
const nameCharAt = name.charAt(0);
console.log(nameCharAt);

// .indexOf() => returns the first occurence of an element
const nameInd = name.indexOf("o")
console.log(nameInd);

// .length
const nameLen = name.length;
console.log(nameLen);

// .trim()
const nameT = name.trim();
console.log(nameT);

// .toUpperCase()
const nameUpperC = name.toUpperCase()
console.log(nameUpperC);

// .toLowerCase()
const nameLowerC = name.toLowerCase()
console.log(nameLowerC);

// .repeat()
const nameR = name.repeat(5);
console.log(nameR);

// StartsWith
const nameSW = name.startsWith("B")
console.log(nameSW);

// .endsWith()
const nameEW = name.endsWith("B")
console.log(nameEW);

// .includes()
const nameIncludes = name.includes("B");
console.log(nameIncludes);

// .replaceAll()
const phoneN = "123-456-7890";
console.log(phoneN.replaceAll("-", ""));
const nameRA = name.replaceAll("o", "a");
console.log(nameRA);

// padStart()
const nameP = name.padStart(15, "-");
console.log(nameP);

// padEnd()
const nameE = name.padEnd(15, "-");
console.log(nameE);

File: section-1/ds/strings/strings.java
public class strings {

    public static void main(String[] args) {
        // STRING = a reference data type that can store one or more characters
        //          reference sata types have access to useful methods
        
        String name = "Shaik Adnan";

        // string.equals() -> case Sensitive
        boolean nameEq = name.equals("Shaik adnan");
        System.out.println(nameEq);  // true if both are same,\
        
        // String.equalsIgnoreCase() -> !Case sensitive
        boolean nameEqIg = name.equalsIgnoreCase("Shaik adnan");
        System.out.println(nameEqIg); // true if both are same,

        // .length()
        int nameLen = name.length();
        System.out.println(nameLen);

        // .charAt()
        char nameChar = name.charAt(6);
        System.out.println(nameChar);

        // .indexOf()
        int nameIndex = name.indexOf("A");
        System.out.println(nameIndex);

        // .isEmpty()
        boolean nameIsEmpty = name.isEmpty();
        System.out.println(nameIsEmpty);

        // .toUpperCase
        String nameUpperC = name.toUpperCase();
        System.out.println(nameUpperC);

        // .toUpperCase
        String nameLowerC = name.toLowerCase();
        System.out.println(nameLowerC);

        // trim()
        String nameT = name.trim();
        System.out.println(nameT);

        // .replace
        String nameReplace = name.replace("a", "o");
        System.out.println(nameReplace);
    }
}